actions.php

```diff
@@ -9,16 +9,21 @@
   // Get the collection.
   $collection = $app->module('collections')->collection($name);
 
-  // Get the collection fields.
+  // Get the collection fields' definitions.
   $fields = [];
   foreach ($collection['fields'] as $field) {
     $fields[$field['name']] = $field;
   }
 
   foreach ($data as $idx => $entry) {
-    foreach ($entry as $fieldName => $values) {
-      // Check for the field type and inspect for presence of styles.
-      switch ($field['type']) {
+    foreach ($entry as $fieldName => $values) {    
+      
+      // Check if current field in entry is defined by user, we don't want '_id' etc. to proceed.
+      if (!isset($fields[$fieldName])) {
+        continue;
+      }
+
+      switch ($fields[$fieldName]['type']) {
         case 'repeater':
           foreach ($data[$idx][$field['name']] as $idx1 => $repeatField) {
```

Comment:
after testing its causing an error on the admin ui for the collections, that can be solved by using instead:

`foreach ((array) $data[$idx][$fieldName] as $idx1 => $repeatField) {`